internal/action/command.go

```diff
@@ -374,6 +374,10 @@ func reloadRuntime(reloadPlugins bool) {
 				}
 			}
 			if _, ok := parsedSettings[k]; ok {
+				if err = config.OptionIsValid(k, parsedSettings[k]); err != nil {
```

Comment:
So... in an attempt to fix bugs, we are adding more special cases in more places, making the code more complicated and messy (as if it wasn't messy already), and undoubtedly adding more hidden bugs to be found in the future. Why, instead, not stop and think, and analyze existing cases, and try to simplify and clean up and generalize them, not increasing but reducing the number of special cases to remember about, to make the code simpler and cleaner and easier to understand and reason about.

1. Currently (without this PR) option validators are only called from `GetNativeValue()`, so they are only called by `SetOption()` and `SetGlobalOption()`, and not called by `SetOptionNative()` or `SetGlobalOptionNative()` called directly, e.g. by a plugin. So why not fix it properly, i.e. move option validation out of `GetNativeValue()` to a proper place, i.e. to `SetOptionNative()` and `SetGlobalOptionNative()`, so that it is automatically done in both cases?

2. Assuming that we can validate options in `config.ReadSettings()`, it would mean that the settings returned by `config.ParsedSettings()` are always already validated, so we don't need to validate them again here. So it seems we can simplify the code organization around `SetGlobalOptionNative()`: we can just have the standard public `SetGlobalOptionNative()` function which does all it should (i.e. validates, checks for locals, writes to `settings.json`), and a low-level private `setGlobalOptionNative()` function (without any extra args) which does none of that, it just sets the global option, - and call that low-level function here in `reload`. (BTW maybe a better name for it would be e.g. `doSetGlobalOption()`, to make it more distinguishable.)

Reply:
Now hopefully much less messy, less error prone and still working.
The initial API is restored again, to not break with the plugins.

Reply:
> 2. [...]

Introduced a further regression which needs to be solved. Since `doSetGlobalOptionNative()` just sets the global option it doesn't touch the buffer locals any longer and thus the most simple case fails again.
Will be solved with the next round, but not today...

Reply:
Hmm, did it? `doSetGlobalOptionNative()` doesn't touch buffer locals, but after that we independently call `b.DoSetOptionNative()`.

In my tests, buffer locals are properly updated after `reload` (e.g. after changing `hlsearch` from true to false in `settings.json` and running `reload`, search highlighting disappears). Do you have different observations?

Reply:
Aha, it seems now I see what you mean... Indeed, we don't properly update buffer locals, since we call `b.DoSetOptionNative()` with the *current* value in `b.Settings`, which is not guaranteed to be updated, it is only updated as a side effect of updating `filetype`. (In particular, `filetype` itself is not properly updated, since it avoids updating itself.) And we are considering removing this side effect of `filetype` from `reload` anyway, so we should fix it properly.

Is this the issue you meant?

Reply:
Yes, this was what I was talking about, since I had already modified my local version in regards of that.
Unfortunately I realized after the comment, that it's revealed after that change earliest.